index.ts

```diff
@@ -0,0 +1,21 @@
+interface PolicyResult {
+  [key: string]: string[];
+}
+
+export = function (policy: string) {
+  return policy.split(';').reduce<PolicyResult>((result, directive) => {
+    const trimmed = directive.trim();
+    if (!trimmed) {
+      return result;
+    }
+
+    const split = trimmed.split(/\s+/g);
+    const key = split.shift() as string;
```

Comment:
Why do you have to specify that this is a `string` here? (This is more of a TypeScript question than an issue with your code.)

Reply:
`Array.prototype.shift` has a return type of `string | undefined`.

If we didn't cast it explicitly to `string` , then we wouldn't be able to do:

```ts
if (!Object.prototype.hasOwnProperty.call(result, key)) {
  result[key] = split;
}
```

since `key` could be `undefined`, which is a type error (`hasOwnProperty` expects the second parameter to be a `string`).

In this case, we can safely make this cast due to knowing that there is at least one non-whitespace letter in the string, thanks to the combination of the `.trim()` and conditional above.